youtube_dl/extractor/common.py

```diff
@@ -2194,7 +2194,11 @@ def geo_verification_headers(self):
         return headers
 
     def _generic_id(self, url):
-        return compat_urllib_parse_unquote(os.path.splitext(url.rstrip('/').split('/')[-1])[0])
+        id = compat_urllib_parse_unquote(os.path.splitext(url.rstrip('/').split('/')[-1])[0])
+        # If our id matches a stopword, use the 2nd-to-last URL path component instead
+        if id in ['story']:
+            id = compat_urllib_parse_unquote(os.path.splitext(url.rstrip('/').split('/')[-2])[0])
```

Comment:
It seems like `story` is always a bad choice for an id, just like `video` would be a bad choice, or classic english stopwords like `the`, `a`, or `an`.
Note that the proposed patch doesn't disallow it, per se, it just chooses another portion of the URL path instead.

Honestly, this seems like a pretty bad way to design a website. I can't figure out why anyone would do it that way. Still, some UK newspapers seem to use this pattern too:
http://www.derbytelegraph.co.uk/quiz-test-your-derby-county-knowledge-as-the-rams-host-bristol-city/story-30126709-detail/story.html
http://www.stokesentinel.co.uk/stoke-on-trent-central-candidates-attend-first-public-hustings/story-30115329-detail/story.html

I did check and there are no tests in any extractor that expect an id of 'story'.

Anyhow, whatever, it seems to me it's better with this than without this, but it doesn't affect core functionality and you disagree, so that's that. And it's currently out of the branch.